lib/services/role.go

```diff
@@ -995,28 +995,75 @@ func (set RoleSet) EnumerateDatabaseUsers(database types.Database, extraUsers ..
 	return result
 }
 
-// EnumerateServerLogins works on a given role set to return a minimal description of allowed set of logins.
-// The wildcard selector is ignored, since it is now allowed for server logins
-func (set RoleSet) EnumerateServerLogins(server types.Server) EnumerationResult {
-	result := NewEnumerationResult()
+// getAllAllowedServerLogins gets all the allowed server
+// logins for the RoleSet.
+func (set RoleSet) getAllAllowedServerLogins() []string {
```

Comment:
This does 3 or 4 unnecessary iterations and could be made a lot quicker by using maps for allowed and denied. WDYT?

Reply:
Something like this?

```go
// getAllAllowedServerLogins gets all the allowed server
// logins for the RoleSet.
func (set RoleSet) getAllAllowedServerLogins() []string {
	mapped := make(map[string]bool)
	for _, role := range set {
		for _, login := range role.GetLogins(types.Allow) {
			mapped[login] = true
		}
		for _, login := range role.GetLogins(types.Deny) {
			mapped[login] = false
		}
	}

	var filtered []string
	for login, isAllowed := range mapped {
		if isAllowed {
			filtered = append(filtered, login)
		}
	}

	return filtered
}
```

Reply:
that'll do

Reply:
e6bec89720a7f6defd113d395a4351a25b393afc

Reply:
(@zmb3 still needs your approval to pass CI)